akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -107,8 +105,8 @@ private[akka] class RemoteWatcher(
 
   val selfHeartbeatRspMsg = HeartbeatRsp(AddressUidExtension(context.system).addressUid)
 
-  // actors that this node is watching, tuple with (watcher, watchee)
-  var watching: Set[(ActorRef, ActorRef)] = Set.empty
+  // actors that this node is watching, map of watchee -> Set(watchers)
+  var watching: Map[InternalActorRef, Set[InternalActorRef]] = Map.empty
```

Comment:
TypedMultiMap is used for mapping types to values.
What is needed is a multimap. There is scala.collection.mutable.MultiMap, but only for mutable maps. I think this is a viable option, though